### ISSUE TITLE
Can we have rounded shadows?

### ISSUE THREAD

**Issue by lillemets** [jonaburg/picom#38]
Currently with rounded corners, shadows are sill square. So the visual effect looks broken. Is there any way to have rounded shadows as well?

![210209_174607](https://user-images.githubusercontent.com/16861099/107389346-61814a80-6aff-11eb-83c3-48af2fd425e2.jpg)

Rounded shadows are already supported by [sdhand's fork](https://github.com/sdhand/picom) but that doesn't support kawase blur.

### Configuration:
```
# Shadow
shadow = true;
shadow-radius = 20;
shadow-offset-x = -20;
shadow-offset-y = -20;
shadow-opacity = 0.5;
shadow-exclude = [
#	"window_type = 'dock'",
  "class_g = 'i3-frame'"
];

# Opacity
inactive-opacity = 1;
active-opacity = 1;
frame-opacity = 1;
inactive-opacity-override = false;
inactive-dim = 0;

# Blur
blur-method = "dual_kawase";
blur-size = 20;
blur-background-exclude = [
#	"window_type = 'dock'",
# "window_type = 'desktop'"
];

# Rounded
corner-radius = 12;
rounded-corners-exclude = [
	"window_type = 'dock'",
  "class_g = 'i3-frame'"
];

# Fading
fading = true;
fade-delta = 2; 
fade-in-step = 0.02;
fade-out-step = 0.02;
no-fading-openclose = false;
fade-exclude = [];

# Other
backend = "glx";
xinerama-shadow-crop = true;
wintypes: {
    dock = { full-shadow = true; }
}

```

**Comment by Axarva** [comment]
I faced this same issue in ibhagwan’s fork of picom as well... no clue yet on what’s going on

**Comment by thorlucas** [comment]
I would also like to see this.

**Comment by hrqmonteiro** [comment]
Yes this is awful, i am not using this fork just because of this. Drives me crazy.

**Comment by lillemets** [comment]
After many months of struggle I finally discovered that this effect is caused by the `--experimental-backends` flag when running Picom.

**Comment by hrqmonteiro** [comment]
> After many months of struggle I finally discovered that this effect is caused by the `--experimental-backends` flag when running Picom.

If i don't use it, it's the same thing for me. The shadows are still now rounded.

**Comment by paulhersch** [comment]
> After many months of struggle I finally discovered that this effect is caused by the `--experimental-backends` flag when running Picom.

have you been using the normal picom version?

**Comment by lillemets** [comment]
> > After many months of struggle I finally discovered that this effect is caused by the `--experimental-backends` flag when running Picom.
> 
> have you been using the normal picom version?

I tried the yshui/picom fork and several others with some different configurations. The only thing that fixed the problem was removing that flag. Unfortunately, without the flag this fork here will lose the transitions, which are kind of the entire point of it. I've now switched to another fork.

**Comment by paulhersch** [comment]
>  I've now switched to another fork.
understandable in your case. I dont use rounded corners and basically jumped from vanilla picom-git to this one for the transitions. I hope the transitions get merged some time tho (as well as Your rounded shadows)

**Comment by hrqmonteiro** [comment]
> > > After many months of struggle I finally discovered that this effect is caused by the `--experimental-backends` flag when running Picom.
> > 
> > 
> > have you been using the normal picom version?
> 
> I tried the yshui/picom fork and several others with some different configurations. The only thing that fixed the problem was removing that flag. Unfortunately, without the flag this fork here will lose the transitions, which are kind of the entire point of it. I've now switched to another fork.

Which fork, may i ask?

**Comment by alba4k** [comment]
Theoretically, [this](https://github.com/Arian8j2/picom-jonaburg-fix) fixes that issue, but it's now unmaintained and really, really old
